zfec/filefec.py

```diff
@@ -251,7 +253,7 @@ def decode_from_files(outf, infiles, verbose=False):
         m = nm
         if not (k is None or k == nk):
             raise CorruptedShareFilesError("Share files were corrupted -- share file %r said that k was %s but another share file previously said that k was %s" % (f.name, nk, k,))
-        if k > len(infiles):
+        if not (k is None or k <= len(infiles)):
```

Comment:
this looks like a bugfix (not related to py3 porting), so likely it should live in a separate commit or even separate PR.